autopilot/agent.go

```diff
@@ -606,22 +607,52 @@ func (a *Agent) openChans(availableFunds btcutil.Amount, numChans uint32,
 			continue
 		}
 
-		// Track the available funds we have left.
-		if availableFunds < chanSize {
-			chanSize = availableFunds
+		for {
+			// Get the fee estimated we need to open this channel
+			// in addition to the already selected ones.
+			feeEstimate, err := a.cfg.ChanController.FeeEstimate(
+				append(chanSizes, chanSize),
+			)
+			if err != nil {
+				return err
+			}
+
+			// Give us some wiggle room.
+			feeEstimate *= 3
```

Comment:
FWIW this can result in a very small channel (right above the min channel size) being created compared to the "fund all" approach.